# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc23/tls-prelink-defines-ppc.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- glibc-2.3.2/elf/elf.h.orig	2003-07-09 02:27:23.000000000 +0200
+++ glibc-2.3.2/elf/elf.h	2003-07-09 02:28:36.000000000 +0200
@@ -1881,8 +1881,40 @@
 #define R_PPC_SECTOFF_LO	34
 #define R_PPC_SECTOFF_HI	35
 #define R_PPC_SECTOFF_HA	36
+
+
+/* Relocations for TLS support */
+#define R_PPC_TLS              67
+#define R_PPC_DTPMOD32         68
+#define R_PPC_TPREL16          69
+#define R_PPC_TPREL16_LO       70
+#define R_PPC_TPREL16_HI       71
+#define R_PPC_TPREL16_HA       72
+#define R_PPC_TPREL32          73
+#define R_PPC_DTPREL16         74
+#define R_PPC_DTPREL16_LO      75
+#define R_PPC_DTPREL16_HI      76
+#define R_PPC_DTPREL16_HA      77
+#define R_PPC_DTPREL32         78
+#define R_PPC_GOT_TLSGD16      79
+#define R_PPC_GOT_TLSGD16_LO   80
+#define R_PPC_GOT_TLSGD16_HI   81
+#define R_PPC_GOT_TLSGD16_HA   82
+#define R_PPC_GOT_TLSLD16      83
+#define R_PPC_GOT_TLSLD16_LO   84
+#define R_PPC_GOT_TLSLD16_HI   85
+#define R_PPC_GOT_TLSLD16_HA   86
+#define R_PPC_GOT_TPREL16      87
+#define R_PPC_GOT_TPREL16_LO   88
+#define R_PPC_GOT_TPREL16_HI   89
+#define R_PPC_GOT_TPREL16_HA   90
+#define R_PPC_GOT_DTPREL16     91
+#define R_PPC_GOT_DTPREL16_LO  92
+#define R_PPC_GOT_DTPREL16_HI  93
+#define R_PPC_GOT_DTPREL16_HA  94
+
 /* Keep this the last entry.  */
-#define R_PPC_NUM		37
+#define R_PPC_NUM              95
 
 /* PowerPC64 relocations defined by the ABIs */
 #define R_PPC64_NONE    R_PPC_NONE
